rules/no-unnecessary-polyfills.js

```diff
@@ -0,0 +1,248 @@
+'use strict';
+const semver = require('semver');
+const readPkgUp = require('read-pkg-up');
+const coreJsCompat = require('core-js-compat');
+const {camelCase, upperFirst} = require('lodash');
+const {STATIC_REQUIRE_SOURCE_SELECTOR} = require('./selectors/index.js');
+
+const {data: compatData, entries: coreJsEntries} = coreJsCompat;
+
+const MESSAGE_ID_POLYFILL = 'unnecessaryPolyfill';
+const MESSAGE_ID_CORE_JS = 'unnecessaryCoreJsModule';
+const messages = {
+	[MESSAGE_ID_POLYFILL]: 'Use the built-in `{{featureName}}`.',
+	[MESSAGE_ID_CORE_JS]: 'All polyfilled features imported from `{{coreJsModule}}` are disponible as built-ins. Use the built-ins instead.',
+};
+
+function getTargetsFromPackage(cwd, useBrowserlistFieldByDefault) {
+	const result = readPkgUp.sync({cwd});
+	if (!result || !result.pkg) {
+		return;
+	}
+
+	const {browserlist} = result.pkg;
+	const nodeEngine = result.pkg.engines && result.pkg.engines.node && new SemverNodeVersion(result.pkg.engines.node);
+	return useBrowserlistFieldByDefault ? browserlist || nodeEngine : nodeEngine || browserlist;
+}
+
+const constructorCaseExceptions = {
+	regexp: 'RegExp',
+};
+function constructorCase(name) {
+	return constructorCaseExceptions[name] || upperFirst(camelCase(name));
+}
+
+const additionalPolyfillPatterns = {
+	'es.promise.finally': '|(p-finally)',
+	'es.object.set-prototype-of': '|(setprototypeof)',
+	'es.string.code-point-at': '|(code-point-at)',
+};
+
+const prefixes = '(mdn-polyfills|polyfill-)';
+const suffixes = '(-polyfill)';
+const delimiter = '(\\.|-|\\.prototype\\.|/)?';
+
+const polyfills = Object.keys(compatData).map(feature => {
+	let [ecmaVersion, constructorName, methodName = ''] = feature.split('.');
+
+	if (ecmaVersion === 'es') {
+		ecmaVersion = `(${ecmaVersion}\\d*)`;
+	}
+
+	constructorName = `(${constructorName}|${camelCase(constructorName)})`;
+	if (methodName) {
+		methodName = `(${methodName}|${camelCase(methodName)})`;
+	}
+
+	const methodOrConstructor = methodName || constructorName;
+
+	const patterns = [
+		`^((${prefixes}?`,
+		`(${ecmaVersion}${delimiter}${constructorName}${delimiter}${methodName}|`, // Ex: es6-array-copy-within
+		`${constructorName}${delimiter}${methodName}|`, // Ex: array-copy-within
+		`${ecmaVersion}${delimiter}${constructorName})`, // Ex: es6-array
+		`${suffixes}?)|`,
+		`(${prefixes}${methodOrConstructor}|${methodOrConstructor}${suffixes})`, // Ex: polyfill-copy-within / polyfill-promise
+		`${additionalPolyfillPatterns[feature] || ''})$`,
+	];
+
+	return {
+		feature,
+		pattern: new RegExp(patterns.join(''), 'i'),
+	};
+});
+
+function report(context, node, feature) {
+	let [, namespace, method = ''] = feature.split('.');
+	if (namespace === 'typed-array' && method.endsWith('-array')) {
+		namespace = method;
+		method = '';
+	}
```

Comment:
> there is no way to know if it is a static method or not

If we can't know, let's just say "use built-in instead", this part seems very risky to me, the module name can be changed in future.